### PR TITLE
[fix](mtmv) Fix mv is deleted in nested mv causing query err and fix some test

### PR DESCRIPTION
**Pull request by seawinde**
### What problem does this PR solve?

if mv2 sql contains mv1 as following, if drop mv_level_1, query mv2 directly would cause err, this fix this
```sql
        CREATE MATERIALIZED VIEW mv2
        BUILD IMMEDIATE REFRESH COMPLETE ON MANUAL
        DISTRIBUTED BY RANDOM BUCKETS 2
        PROPERTIES ('replication_num' = '1') 
        AS
select * from mv_level_1;
```
Issue Number: close #xxx

Related PR: #45045

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Thearas** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

File: regression-test/suites/auth_p0/test_select_column_auth.groovy [hunk 1]
```diff
@@ -130,6 +130,9 @@ suite("test_select_column_auth","p0,auth") {
     sql """grant select_priv(sum_id) on ${dbName}.${mtmv_name} to ${user}"""
     sql """grant select_priv(id) on ${dbName}.${tableName} to ${user}"""
     connect(user, "${pwd}", context.config.jdbcUrl) {
+        logger.info("show grants;")
```
Comment (zddr) [inline regression-test/suites/auth_p0/test_select_column_auth.groovy:133]: Only print this string?

File: regression-test/suites/mv_p0/unique/unique_rewrite.groovy [hunk 2]
```diff
@@ -96,6 +96,10 @@ suite("mv_on_unique_table") {
     AS
     ${mv1}
     """)
+
+    def desc_all_mv1 = sql """desc lineitem_2_uniq all;"""
+    logger.info("desc mv1 is" + desc_all_mv1)
```
Comment (zddr) [inline regression-test/suites/mv_p0/unique/unique_rewrite.groovy:101]: not need `desc_all_mv1.toString()`?

File: regression-test/suites/nereids_rules_p0/mv/is_in_debug_mode/is_in_debug_mode.groovy [hunk 3]
```diff
@@ -115,7 +119,9 @@ suite("is_in_debug_mode") {
         AS select * from orders where o_orderkey > 2;
         """
     } catch (Exception e) {
-        Assert.assertTrue(e.getMessage().contains("because is in debug mode"))
+        def message = e.getMessage()
+        logger.info("test_create_mv3" + message)
```
Comment (zddr) [inline regression-test/suites/nereids_rules_p0/mv/is_in_debug_mode/is_in_debug_mode.groovy:123]: suggest:  logger.info("test_create_mv3: " + message)

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by zddr — APPROVED** [review]

**Pull request merged**